addons/sale/models/sale_order.py

```diff
@@ -1301,7 +1301,8 @@ def _notify_get_recipients_groups(self, message, model_description, msg_vals=Non
             pass
         else:
             access_opt = customer_portal_group[2].setdefault('button_access', {})
-            is_tx_pending = self.get_portal_last_transaction().state == 'pending'
+            # Only to read tx, does not produce a sudoed recordset and no information is leaked.
```

Comment:
It's obvious when you read the code. The comment should justify using a sudo so we know why it's there, not explain that it has no consequence.
```suggestion
            #  In sudo mode to allow salesmen to read the state of the transaction.
```